addon/ajax-request.js

```diff
@@ -148,6 +164,17 @@ export default class AjaxRequest {
   }
 
   /**
+   * @method _getFullHeadersHash
+   * @private
+   * @param {Object} headers
+   * @return {Object}
+   */
+  _getFullHeadersHash(headers) {
+    const classHeaders = get(this, 'headers') || {};
+    return merge(classHeaders, headers);
```

Comment:
That's fine too, just not sure why that's necessary over merging `classHeaders` and `headers` directly.  Would mutating `classHeaders` actually change the value of `this.headers`?